lib/active_model/serializer/collection_serializer.rb

```diff
@@ -73,8 +72,9 @@ def serializers_from_resources
       def serializer_from_resource(resource, serializer_context_class, options)
         serializer_class = options.fetch(:serializer) { serializer_context_class.serializer_for(resource) }
 
-        if serializer_class.nil? # rubocop:disable Style/GuardClause
-          fail NoSerializerError, "No serializer found for resource: #{resource.inspect}"
+        if serializer_class.nil?
+          ActiveModelSerializers.logger.debug "No serializer found for resource: #{resource.inspect}"
```

Comment:
If we merge this, it fixes https://github.com/rails-api/active_model_serializers/pull/1693 I believe cc @beauby